### PR TITLE
Add context manager for printing prefixed stack trace

### PR DESCRIPTION
**Pull request by dkmiller**
Add functionality for wrapping potentially problematic code in a Python `with` statement (easier than explicit try / except).

## To-do

- [x] Once design and implementation is approved, add examples.

### CODE REVIEW

File: src/confidential_ml_utils/exceptions.py [hunk 1]
```diff
@@ -145,3 +145,29 @@ def wrapper(*func_args, **func_kwargs):
...
+        self.disable = disable
+        self.prefix = prefix
+        self.scrub_message = scrub_message
+        self.keep_message = keep_message
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, value, tb):
+        foo = str(bool(exc_type))
```
Comment (pjanowski) [inline src/confidential_ml_utils/exceptions.py:169]: What is this for?

File: src/confidential_ml_utils/exceptions.py [hunk 2]
```diff
@@ -145,3 +145,29 @@ def wrapper(*func_args, **func_kwargs):
...
+        self.file = file
+        self.disable = disable
+        self.prefix = prefix
+        self.scrub_message = scrub_message
+        self.keep_message = keep_message
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, value, tb):
```
Comment (pjanowski) [inline src/confidential_ml_utils/exceptions.py:168]: Just curious, how to you pass the build checks, i.e. won't flake complain about unused variables? Also, exc_value, traceback would be a bit more readable.

File: src/tests/test_exceptions.py [hunk 3]
```diff
@@ -85,6 +86,21 @@ def function():
...
+def test_prefix_stack_trace_respects_scrub_message(message):
+    """
+    Verify that the "message scrubbed" string added in by `prefix_stack_trace`
+    respects the provided configuration.
+    """
+    file = io.StringIO()
+
+    with pytest.raises(Exception):
+        with PrefixStackTrace(scrub_message=message, file=file):
+            raise Exception(message)
```
Comment (pjanowski) [inline src/tests/test_exceptions.py:99]: For consistency with the other tests use a locally defined function to raise the exception?

File: src/tests/test_exceptions.py [hunk 4]
```diff
@@ -85,6 +86,21 @@ def function():
...
+@pytest.mark.parametrize("message", ["foo__"])
+def test_prefix_stack_trace_respects_scrub_message(message):
+    """
+    Verify that the "message scrubbed" string added in by `prefix_stack_trace`
+    respects the provided configuration.
+    """
+    file = io.StringIO()
+
+    with pytest.raises(Exception):
+        with PrefixStackTrace(scrub_message=message, file=file):
```
Comment (pjanowski) [inline src/tests/test_exceptions.py:98]: Using the same string for both the exception message and the prefix. Won't that make the assert potentially weaker? Better to use a different message for the exception.

File: src/confidential_ml_utils/exceptions.py (hunk 1 above)
Comment (dkmiller) [reply src/confidential_ml_utils/exceptions.py:169]: Oops, it should not be there!

File: src/confidential_ml_utils/exceptions.py (hunk 2 above)
Comment (dkmiller) [reply src/confidential_ml_utils/exceptions.py]: I think `flake8` understands that "magic" methods (e.g., `__enter__`, `__str__`) won't necessarily use all arguments.

Good suggestion on variable names.

**Pull request merged**